### ISSUE TITLE
Build fails when the main field targets a sub-directory

### ISSUE THREAD

**Issue by cpv123** [martpie/next-transpile-modules#138]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE

**Are you trying to transpile a local package or an npm package?**
Local package

**Describe the bug**
A clear and concise description of what the bug is (don't forget the logs if applicable).

My local package is not being transpiled when building an application with `yarn build`. This bug only happens when using `next-transpile-modules` version 5.

I have a monorepo built with `yarn workspaces` that contains two NextJS (v10) applications that share code between each other - a link to the repo is below. When running `yarn dev` each application is built and runs fine, but when running `yarn build` it fails because of webpack errors that indicate that the imported code is not being transpiled: `Module parse failed: Unexpected token (1:25)`
 
This only happens with `next-transpile-modules` version 5. If I downgrade to version 4.1.0 then the application builds fine.

**To Reproduce**
Steps to reproduce the behavior (a reproduction is even better):

You can reproduce the issue using this repo + branch: https://github.com/cpv123/nextjs-sharing-code-monorepo/tree/next-transpile-modules-v5.

In the README I've added a small section 'Known issue with next-transpile-modules version 5' which describes the reproduction steps.

**Expected behavior**

`yarn build` should successfully build the application

**Setup**

- Next.js version: 10.0.3
- `next-transpile-modules` version: 5.0.0
- Node.js version: 14.15.0
- `npm`/`yarn` version: 1.21.1
- Operating System: MacOS
- Webpack 4 or 5: 4 (using default version from NextJS)

**Additional context**

* This is happening on both my local machine and in my CI/CD pipeline on Vercel.
* I have followed the upgrade steps for https://github.com/martpie/next-transpile-modules/releases/tag/5.0.0 i.e. adding a `main` field to each `package.json`. This allowed `yarn dev` to work.

**Comment by martpie** [comment]
Hello 👋 

As a temporary fix, you can point your `main` field to an empty `index.js` at the root of your application (I think this will fix it).

The reason why it's not working is because it will only transpile the `dirname` of `main`, so here, only your `pages` folder will get transpiled.

This is not the most straightforward behavior. One thing I could add in the lib is "go up until you find a package.json" and use this path as root. 

More magic, but probably a better default. I need to think a little bit about it.

**Comment by martpie** [comment]
Basically, the question to answer is: "Do some people transpile only parts of their packages"?

If yes, we need to keep it as it is, I can maybe add a "usePackageJsonDir" option or something for people with your case.

If no, I can make it the default (probably a new major version as it is a potential breaking change)

**Comment by martpie** [comment]
@cpv123 I have a beta version ready, would you be interested in testing it?

**Comment by cpv123** [comment]
@martpie thanks for the quick response. 

**Short story:** using your beta version at https://github.com/martpie/next-transpile-modules/tree/resolve-modules-to-their-root-package-json works 🎉 I have left the main fields as they were (`"main": "pages/index.js"`) and now the build works.

**Longer story:** I first tried your temporary fix suggestion - pointing main to an empty index at the root - and the build still failed, but this time for a different reason. I was getting `Minified React error #321` ==> `Hooks can only be called inside of the body of a function component` https://reactjs.org/docs/error-decoder.html?invariant=321. I tried debugging briefly, but as far as I could tell there was only one copy/version of `react` in the entire repo so I didn't figure out what was causing it.

Perhaps this isn't an issue given that your beta version works, but thought I'd let you know anyway.
